src/auto/unix_input_stream.rs

```diff
@@ -0,0 +1,84 @@
+// This file was generated by gir (https://github.com/gtk-rs/gir)
+// from gir-files (https://github.com/gtk-rs/gir-files)
+// DO NOT EDIT
+
+use InputStream;
+use PollableInputStream;
+use gio_sys;
+use glib::object::Cast;
+use glib::object::IsA;
+use glib::signal::SignalHandlerId;
+use glib::signal::connect_raw;
+use glib::translate::*;
+use glib_sys;
+use std::boxed::Box as Box_;
+use std::fmt;
+use std::mem::transmute;
+
+glib_wrapper! {
+    pub struct UnixInputStream(Object<gio_sys::GUnixInputStream, gio_sys::GUnixInputStreamClass, UnixInputStreamClass>) @extends InputStream, @implements PollableInputStream;
+
+    match fn {
+        get_type => || gio_sys::g_unix_input_stream_get_type(),
+    }
+}
+
+impl UnixInputStream {
+    pub fn new(fd: i32, close_fd: bool) -> UnixInputStream {
```

Comment:
This has to be via `IntoRawFd` and `unsafe` instead of just taking a random integer as argument for the file descriptor. See `gio::Socket` bindings.

Reply:
Could you explain why this needs to be `unsafe` if it takes a `IntoRawFd`?

AFAICT, `IntoRawFd` is defined as taking `self` (without &), so the call to `into_raw_fd` should move the src, meaning nothing else can use the fd at that point, right? Is there something else about the call that is unsafe?

I feel like to satisfy Rust's safety constraints for RawFd using classes, the close_fd parameter doesn't really make sense to expose this way. Maybe via a different function that takes an AsRawFd.

Reply:
You can do the following with completely safe code to generate invalid fds from random numbers:

```rust
struct Meh(i32);

impl IntoRawFd for Meh {
  fn into_raw_fd(self) -> RawFd { self.0 }
}

let fd: RawFd = Meh(123).into_raw_fd();
```

So any function taking a `RawFd` must be marked as unsafe as the caller need to ensure that the passed value is valid and ok to use in this context.


For `close_fd`, I agree.